tools/new_module.php

```diff
@@ -0,0 +1,359 @@
+#!/usr/bin/env php
+<?php declare(strict_types=1);
+/**
+ * PHP Version 8
+ *
+ * @author  Dave MacFarlane <dave.macfarlane@mcin.ca>
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link    https://www.github.com/aces/Loris/
+ */
+
+require_once 'generic_includes.php';
+
+$flagsEnd = 0;
+$flags    = getopt("hj", ['jsx', 'help'], $flagsEnd);
+
+
+$jsx  = isset($flags['j']) || isset($flags['jsx']);
+$help = isset($flags['h']) || isset($flags['help']);
+
+if ($help) {
+    usage();
+    exit(0);
+}
+if ($argc !== $flagsEnd+1) {
+    usage();
+    exit(1);
+}
+$mdir = $argv[$flagsEnd];
+
```

Comment:
if user run from the 'tools' folder, which means it will check for the module name in the current tools folder. However, for best results, we recommend running the script from the 'modules' folder.  It is advisable to remind users to run it from the 'modules' folder.

Reply:
I'm not sure what you mean. It checks the existence of the directory given as an argument to be sure it's not overwriting something. It uses the last part of the path name as the module name. What does it matter where it's run from?